templates/default/coredns-configmap.yml.erb

```diff
@@ -22,7 +22,12 @@ data:
         loop
         reload
         loadbalance
-    } 
-    in-addr.arpa ip6.arpa { 
+    }
+    <%= node['consul']['domain'] %>:53 {
```

Comment:
I don't understand. Why redirecting only the consul domain here and not everything? I think it would allow us to get rid of the `fallback_dns` attribute which create tons of confusion.

Reply:
`/etc/resolv.conf` should be able to resolv everything (?)

Reply:
actually if you do, keep in mind that `in-addr.arpa` and `ip6.arpa` are needed to do a reverse dns lookup. So I'd probably redirect them as well on `/etc/resolv.conf`

Reply:
I've tried that. For some reason it's killing dnsmasq

```
[root@ip-10-0-1-16 centos]# journalctl -f -u dnsmasq
-- Logs begin at Tue 2020-04-07 06:15:46 UTC. --
Apr 07 07:41:06 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:41:16 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:41:26 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:41:36 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:41:46 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:41:56 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
Apr 07 07:42:06 ip-10-0-1-16.eu-west-1.compute.internal dnsmasq[17631]: Maximum number of concurrent DNS queries reached (max: 150)
```

I increased the limit but it's been reached as soon as I make a DNS query from a pod. Forwarding only service discovery names was the only way to resolve them.

Reply:
Ok then